File: src/CrowdfindingApp.Client/src/app/models/replies/rewards/RewardInfo.ts
import { GenericLookupItem } from '../../common/GenericLookupItem';
import { DeliveryTypeEnum } from '../../enums/DeliveryTypeEnum';
import { RewardCard } from 'src/app/components/reward-card/RewardCard';

export class RewardInfo {
  public id: string;
  public projectId: string;
  public title: string;
  public price: number;
  public description: string;
  public deliveryDate: Date;
  public isLimited: boolean;
  public limit: number;
  public image: string;
  public deliveryType: DeliveryTypeEnum;
  public deliveryCountries: GenericLookupItem<string, number>[];

  public getCard = function() {
    return new RewardCard(
      this.title,
      this.price,
      this.description,
      this.image,
      this.deliveryType,
      this.deliveryDate,
      this.limit
    );
  }
  // getCard(): RewardCard {
  //   return new RewardCard(
  //     this.title,
  //     this.price,
  //     this.description,
  //     this.image,
  //     this.deliveryType,
  //     this.deliveryDate,
  //     this.limit
  //   );
  // }
}

File: src/CrowdfundingApp.Common.Core/DataTransfers/Rewards/RewardInfo.cs
using System;
using System.Collections.Generic;
using CrowdfindingApp.Common.Enums;

namespace CrowdfindingApp.Common.Core.DataTransfers.Rewards
{
    public class RewardInfo
    {
        public string Id { get; set; }
        public string ProjectId { get; set; }
        public string Title { get; set; }
        public decimal? Price { get; set; }
        public string Description { get; set; }
        public DateTime? DeliveryDate { get; set; }
        public bool IsLimited { get; set; }
        public int? Limit { get; set; }
        public string Image { get; set; }
        public DeliveryType? DeliveryType { get; set; }
        public List<KeyValue<string, decimal?>> DeliveryCountries { get; set; }
    }
}
